core/src/main/java/org/keycloak/jose/jws/JWSBuilder.java

```diff
@@ -74,6 +76,7 @@ protected String encodeHeader(String sigAlgName) {
 
         if (type != null) builder.append(",\"typ\" : \"").append(type).append("\"");
         if (kid != null) builder.append(",\"kid\" : \"").append(kid).append("\"");
+        if (thumbprint != null) builder.append(",\"x5t\" : \"").append(thumbprint).append("\"");
```

Comment:
Is this related to exposing more info in admin UI?

If we are about to add support for adding the thumbprint to the JWS header, we should probably consider adding `x5t#S256`.